source/material_model/steinberger.cc

```diff
@@ -237,19 +237,23 @@ namespace aspect
           unsigned int layer_index = std::distance(conductivity_transition_depths.begin(),
                                                    std::lower_bound(conductivity_transition_depths.begin(),conductivity_transition_depths.end(), depth));
 
-          // Make sure the temperature is positive.
-          const double T = std::max(temperature, 1.e-3 * conductivity_reference_temperatures[layer_index]);
-          const double T_dependence = std::pow(conductivity_reference_temperatures[layer_index] / T, conductivity_exponents[layer_index]);
           const double p_dependence = reference_thermal_conductivities[layer_index] + conductivity_pressure_dependencies[layer_index] * pressure;
 
+          // Make reasonably sure we will not compute any invalid values due to the temperature-dependence.
+          // Since both the temperature-dependence and the saturation term scale with (Tref/T), we have to
+          // make sure we can compute the square of this number. If the temperature is small enough to
+          // be close to yielding NaN values, the conductivity will be set to the maximum value anyway.
+          const double T = std::max(temperature, sqrt(std::numeric_limits<double>::min() * conductivity_reference_temperatures[layer_index]));
```

Comment:
Yes, this looks better, but I think you meant this:
```suggestion
          const double T = std::max(temperature, std::sqrt(std::numeric_limits<double>::min()) * conductivity_reference_temperatures[layer_index]);
```
In the current form if conductivity_reference_temperatures < 1.0 the square root would be 0.

Ready to go otherwise.